extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java

```diff
@@ -1,145 +1,153 @@
+import org.json.JSONWriter;
+import com.google.refine.commands.Command;
 
 /**
  * Enables automatic file format detection
+ * 
  * @author Shamanou van Leeuwen
  * @date May 24 2017
  *
  */
 
 
 /**
- * A command which detects the file format, should be configured in the controller 
- * and called from the javascript front-end.
+ * A command which detects the file format, should be configured in the controller and called from
+ * the javascript front-end.
  * 
  */
-public class DetectFileFormatCommand extends Command{
+public class DetectFileFormatCommand extends Command {
     RDFParser rdfParserttl = new TurtleParser();
     RDFParser rdfParserntriples = new NTriplesParser();
+    RDFParser rdfParserrdfxml = new RDFXMLParser();
+    RDFParser rdfParserJsonld = new JSONLDParser();
+    RDFParser rdfParserNQuads = new NQuadsParser();
+    RDFParser rdfParserJson = new RDFJSONParser();
+    RDFParser rdfParserTrix = new TriXParser();
+    RDFParser rdfParserTrig = new TriGParser();
 
     RDFParser[] parsers;
 
     /**
-     * This method takes a request containing rdf and tries to iterate over 
-     * different parsers to see which one is able to parse the data.
-     * If it is able to be parsed a format string is returned by passing it 
-     * to the response object in JSON-format. 
+     * This method takes a request containing rdf and tries to iterate over different parsers to see
+     * which one is able to parse the data. If it is able to be parsed a format string is returned
+     * by passing it to the response object in JSON-format.
      * 
      * When there is no format detected the parser defaults to rdf/xml.
      * 
-     * Other formats that can be parsed are Turtle and Ntripples.
-     * NOTE: Sesame supports more formats but the currently used version 
-     * gave problems with other parsers. 
+     * Other formats that can be parsed are Turtle and Ntripples. NOTE: Sesame supports more formats
+     * but the currently used version gave problems with other parsers.
      * 
      * @param req a request object
      * @param res a response object
-     * @see com.google.refine.commands.Command#doPost(javax.servlet.http.HttpServletRequest, javax.servlet.http.HttpServletResponse)
+     * @see com.google.refine.commands.Command#doPost(javax.servlet.http.HttpServletRequest,
+     *      javax.servlet.http.HttpServletResponse)
      */
     @Override
-    public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
-        
-        this.parsers = new RDFParser[]{this.rdfParserntriples, this.rdfParserttl};
+    public void doPost(HttpServletRequest req, HttpServletResponse res)
+            throws ServletException, IOException {
+
+        this.parsers = new RDFParser[] {this.rdfParserntriples, this.rdfParserttl,
+                this.rdfParserrdfxml, this.rdfParserJsonld, this.rdfParserNQuads,
+                this.rdfParserJson, this.rdfParserTrix, this.rdfParserTrig};
```

Comment:
This array will be initialized every single time this command is invoked. Doesn't seem to be the intent? Or at least it should't be.